shared/utils/shuffle-array.ts

```diff
@@ -4,6 +4,12 @@ export const shuffleArray = <T>(arrToShuffle: Array<T>) => {
 
   for (let i = arr.length - 1; i > 0; i--) {
     const j = Math.floor(Math.random() * (i + 1));
+
+    // tsc would throw an error here as it can't guarantee that
+    // the elements being swapped would have the same type.
+    // Type check can be ignored here since the function isn't interested in the element values.
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```

Comment:
```suggestion
    // We know that i and j are within the bounds of the array, TS doesn't
    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
```
The error shows up because the api's config is stricter than the shared folder's config. I'm not sure how to fix this, since VSCode (correctly) uses the nearest config (shared).

Anyway, I think this clarifies the error without needing to resort to `@ts-ignore`